### PR TITLE
Cypress Setup for E2E testing

### LINKED ISSUE #111: Setup Cypress for E2E testing
**Is your enhancement related to a problem? Please describe.**

See overall concept in the [OSBP guide](https://10up.github.io/Open-Source-Best-Practices/testing/#e2e-testing) and example setups on [Simple Podcasting](https://github.com/10up/simple-podcasting), [Restricted Site Access](https://github.com/10up/restricted-site-access), and https://github.com/10up/ElasticPress/pull/2446.

**Describe the solution you'd like**

- [x]   add base GitHub Action
- [x]   add .wp-env.json
- [x]   add base docker files
- [x]   update package.json / package lock files
- [x]   add cypress tests
- [x]   remove any WP Acceptance components

**Designs**

n/a

**Describe alternatives you've considered**

n/a

**Additional context**

https://github.com/10up/cypress-wp-setup/ && https://github.com/10up/cypress-wp-utils will be helpful in getting this spun up and some initial test commands to leverage.

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change

* Cypress tests setup with the help/use of https://github.com/10up/cypress-wp-setup/ & https://github.com/10up/cypress-wp-utils.
* Also adds a condition while using `wp_get_original_image_path()` to support minimum WP version (i.e. `4.6`) and instead used `get_attached_file` while clearing the avatar cache.

Closes #111.

### Alternate Designs

n/a

### Possible Drawbacks

n/a

### Verification Process

Make sure all tests added are running without fail.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Added - Cypress tests for E2E testing.

### Credits

Props @faisal-alvi @vikrampm1

### CODE REVIEW

File: tests/cypress/integration/avatars-migration.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+describe('Check if avatar migration works', () => {
```
Comment (Sidsector9) [inline tests/cypress/integration/avatars-migration.test.js]: Although the title implies we are testing if migration works, the test itself doesn't perform any migration.
I think we can change it to:

```js
describe( 'Avatar migration', () => {} )
```
As the context for this test is migration of avatars.

And the test title can be set to:

```js
it( 'Test no migration from WP User Avatar' );

File: tests/cypress/integration/set-avatar.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import 'cypress-file-upload';
+
+describe('Check if admin can upload, crop and select local avatar', () => {
+    before(() => {
+        cy.login();
+    });
+
+    it('Can admin upload, crop and select local avatar?', () => {
```
Comment (Sidsector9) [inline tests/cypress/integration/set-avatar.test.js:8]: As we're not testing cropping in this test, it can be removed from the title.

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

File: tests/cypress/integration/set-avatar.test.js (hunk 2 above)
Comment (faisal-alvi) [reply tests/cypress/integration/set-avatar.test.js:8]: Actually, we are, the cropping action runs so fast that we don't see it:

sla-pr-115

Please let me know if this is not what you expected.

Comment (Sidsector9) [reply tests/cypress/integration/set-avatar.test.js:8]: brilliant! 👍

**Review by Sidsector9 — APPROVED** [review]

**Pull request merged**